core/computed/main-resource.js

```diff
@@ -22,7 +22,16 @@ class MainResource {
     const {mainDocumentUrl} = data.URL;
     if (!mainDocumentUrl) throw new Error('mainDocumentUrl must exist to get the main resource');
     const requests = await NetworkRecords.request(data.devtoolsLog, context);
-    const mainResource = NetworkAnalyzer.findResourceForUrl(requests, mainDocumentUrl);
+
+    const mainResourceRequests = requests.filter(request =>
```

Comment:
1- Should we add a new function to `NetworkAnalyzer` for `findLastResourceForUrl`?

2- I got rid of the `resourceUrl.startsWith(request.url)` optimization check because my test case tripped up how it may be broken. I think it might need to become `(resourceUrl.startsWith(request.url) || request.url.startsWith(resourceUrl))` to be correct?

Reply:
> Should we add a new function to NetworkAnalyzer for findLastResourceForUrl?

I think this is a good idea.

> I got rid of the resourceUrl.startsWith(request.url) optimization check because my test case tripped on it, suggesting that it may be broken.

What did it trip on?

Reply:
Two urls can be equal ignoring the fragment, without one being a substring of the other.

Reply:
`request.url` won't ever have fragments, so it should always be the start of a URL that it's equal to when excluding fragments.